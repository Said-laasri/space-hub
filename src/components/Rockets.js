import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getRockets } from '../redux/rockets/rockets';
import Rocket from './Rocket';

let isInitial = true;

function Rockets() {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocket);

  useEffect(() => {
    if (isInitial) {
      dispatch(getRockets());
      isInitial = false;
    }
  }, [dispatch]);

  return (
    <div className="rocket">
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.id}
          name={rocket.name}
          description={rocket.description}
          img={rocket.img}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
}

export default Rockets;
