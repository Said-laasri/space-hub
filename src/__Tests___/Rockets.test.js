import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configueStore';
import Rockets from '../components/Rocket';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Rockets
            key={0}
            id={0}
            reserved={false}
            name="rockets.name"
            description="rockets.description"
            img="rockets.img"
          />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
