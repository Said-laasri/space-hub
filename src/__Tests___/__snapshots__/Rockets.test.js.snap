// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rocket d-flex"
>
  <img
    alt="rocket_image"
    className="rocket-img"
    src="rockets.img"
  />
  <div
    className="rocket-info"
  >
    <h3>
      rockets.name
    </h3>
    <p>
      <span />
      rockets.description
    </p>
    <button
      id={0}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
